### PR TITLE
Add support for using raw disks with xhyve

### PR DESCRIPTION
**Pull request by dalehamel**
WIP - this should be an actual config flag

Depends on https://github.com/zchee/docker-machine-driver-xhyve/pull/169

### CODE REVIEW

**Comment by aaron-prindle** [comment]
@minikube-bot ok to test

**Comment by dalehamel** [comment]
@aaron-prindle FYI this is WIP, not ready to merge. I need to land a patch into docker-machine-driver-xhyve first, then I want to make this configurable instead of hardcoded.

**Comment by dlorenc** [comment]
@dalehamel I think your patch is in now right? What's left here?

**Comment by dalehamel** [comment]
Yes it landed today, I need to fix this PR up to be able to toggle it    
  
  
  
  
  

  
  
>

**Comment by dalehamel** [comment]
@dlorenc @aaron-prindle thoughts on approach for making this configurable? Is this too ugly to expose as a user-configurable? I saw some other driver-specific flags, so I tried to follow the established standard.

**Comment by r2d4** [comment]
I'm ok with it not being user configurable if there are clear performance gains from using a raw disk.

Either way, I think its OK to have a few driver specific flags to minikube start.  Early on we chose not to emulate to the docker-machine dynamic flags per driver for simplicity.

**Comment by dalehamel** [comment]
> I'm ok with it not being user configurable if there are clear performance gains from using a raw disk.

In our experience, there is a substantial difference (nearly 10x when running MySQL in minikube), but I think there's a reasonable argument to be made about stability potentially being a tradeoff. It might make sense to make it a default in the future, but probably best to have people opt-in for now.

> Either way, I think its OK to have a few driver specific flags to minikube start. Early on we chose not to emulate to the docker-machine dynamic flags per driver for simplicity.

Cool, I'll leave this as-is for now, barring fixing a couple of CI code formatting errors I've just noticed.

**Comment by dalehamel** [comment]
Not sure why some of the builds are failing, don't seem related to the PR to me

**Comment by dalehamel** [comment]
@r2d4 @dlorenc @aaron-prindle I could use a hand figuring out why these tests are red

**Comment by dlorenc** [comment]
@minikube-bot test this please

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -217,6 +219,7 @@ func init() {
 	startCmd.Flags().String(hostOnlyCIDR, "192.168.99.1/24", "The CIDR to be used for the minikube VM (only supported with Virtualbox driver)")
 	startCmd.Flags().String(hypervVirtualSwitch, "", "The hyperv virtual switch name. Defaults to first found. (only supported with HyperV driver)")
 	startCmd.Flags().String(kvmNetwork, "default", "The KVM network name. (only supported with KVM driver)")
+	startCmd.Flags().String(diskDriver, "ahci-hd", "The disk driver to use [ahci-hd|virtio-blk] (only supported with xhyve driver)")
```
Comment (dlorenc) [inline cmd/minikube/cmd/start.go]: Would you mind adding "xhyve" to the flag name, similar to the hyperv-virtual-switch one?

File: pkg/minikube/cluster/cluster_darwin.go [hunk 2]
```diff
@@ -69,7 +71,8 @@ func createXhyveHost(config MachineConfig) *xhyveDriver {
 		BootCmd:        "loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 base host=" + cfg.GetMachineName(),
 		DiskSize:       int64(config.DiskSize),
 		Virtio9p:       true,
-		Virtio9pFolder: "/Users",
+		QCow2:          false,
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster_darwin.go]: What happens if we pass these new options to an older version of the xhyve driver? Will it just ignore them?

Comment (dalehamel) [inline pkg/minikube/cluster/cluster_darwin.go]: @dlorenc It should just ignore them https://github.com/zchee/docker-machine-driver-xhyve/blob/master/xhyve/xhyve.go#L226-L250

File: cmd/minikube/cmd/start.go (hunk 1 above)
Comment (dalehamel) [inline cmd/minikube/cmd/start.go]: @dlorenc that's sensible, will do.

**Comment by dalehamel** [comment]
@dlorenc to match the naming with Hyperv, i prefixed all references with Xhyve.

File: pkg/minikube/cluster/cluster_darwin.go [hunk 3]
```diff
@@ -69,7 +71,8 @@ func createXhyveHost(config MachineConfig) *xhyveDriver {
 		BootCmd:        "loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 base host=" + cfg.GetMachineName(),
 		DiskSize:       int64(config.DiskSize),
 		Virtio9p:       true,
-		Virtio9pFolder: "/Users",
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster_darwin.go]: Did you mean to delete this one?

Comment (dalehamel) [inline pkg/minikube/cluster/cluster_darwin.go]: Nope - good catch.

**Comment by dalehamel** [comment]
@dlorenc I think the xhyve test should pass with the latest code

**Comment by dlorenc** [comment]
Yeah, failure looks unrelated. Merging. Thanks!

**Comment by dalehamel** [comment]
@dlorenc awesome! What is the release schedule like?

I don't mean to pester (you literally just merged it), I just want to know when we can move back to using minikube from homebrew instead of our custom build :)

**Pull request merged**